backend/tests/market/test_market.py

```diff
@@ -16,245 +14,279 @@
 
 User = get_user_model()
 
-class TestMarket(TestCase):
 
+class BaseMarketTest(TestCase):
     def setUp(self):
```

Comment:
We ought to maximize the amount of setup that is done here as opposed to copying code between each case. Please go through each case and find as much shared/stuff you can put here as possible :)